examples/window-app/nrfconnect/main/WindowCovering.cpp

```diff
@@ -0,0 +1,340 @@
+/*
+ *    Copyright (c) 2022 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "WindowCovering.h"
+#include "AppConfig.h"
+#include "PWMDevice.h"
+
+#include <dk_buttons_and_leds.h>
+
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app/util/af.h>
+#include <logging/log.h>
+#include <platform/CHIPDeviceLayer.h>
+#include <zephyr.h>
+
+LOG_MODULE_DECLARE(app, CONFIG_MATTER_LOG_LEVEL);
+
+using namespace ::chip::Credentials;
+using namespace ::chip::DeviceLayer;
+using namespace chip::app::Clusters::WindowCovering;
+
+static k_timer sLiftTimer;
+static k_timer sTiltTimer;
+static constexpr uint32_t sMoveTimeoutMs{ 200 };
+
+static constexpr uint32_t FromOneRangeToAnother(uint32_t aInMin, uint32_t aInMax, uint32_t aOutMin, uint32_t aOutMax,
+                                                uint32_t aInput)
+{
+    const auto diffInput  = aInMax - aInMin;
+    const auto diffOutput = aOutMax - aOutMin;
+    if (diffInput > 0)
+    {
+        return static_cast<uint32_t>(aOutMin + static_cast<uint64_t>(aInput - aInMin) * diffOutput / diffInput);
+    }
+    return 0;
+}
+
+WindowCovering::WindowCovering()
+{
+    mLiftLED.Init(LIFT_STATE_LED);
+    mTiltLED.Init(TILT_STATE_LED);
+
+    if (mLiftIndicator.Init(LIFT_PWM_DEVICE, LIFT_PWM_CHANNEL, 0, 255) != 0)
+    {
+        LOG_ERR("Cannot initialize the lift indicator");
+    }
+    if (mTiltIndicator.Init(TILT_PWM_DEVICE, TILT_PWM_CHANNEL, 0, 255) != 0)
+    {
+        LOG_ERR("Cannot initialize the tilt indicator");
+    }
+
+    k_timer_init(&sLiftTimer, MoveTimerTimeoutCallback, nullptr);
+    k_timer_init(&sTiltTimer, MoveTimerTimeoutCallback, nullptr);
+}
+
+void WindowCovering::MoveTimerTimeoutCallback(k_timer * aTimer)
+{
+    if (!aTimer)
+        return;
+
+    if (aTimer == &sLiftTimer)
+    {
+        chip::DeviceLayer::PlatformMgr().ScheduleWork(DriveCurrentLiftPosition);
```

Comment:
Those timer call can have been consolidated with the current common window app 
even if we have to touch the EFR32 demo